### PR TITLE
fix(es/compat): counteract unrolling of awaited async arrays in generator

### PR DESCRIPTION
**Pull request by finnboeger**
**Description:**
This is a workaround addressing the issue #5951 by wrapping the arguments of the `.concat` call in individual arrays to counteract the unrolling done by `concat`. 

A different way of handling the problem would be to `push` the arguments to `_tmp` before the return statement and then use `_tmp` in the `apply` call. However I'm not comfortable enough with the codebase to implement it that way.

**BREAKING CHANGE:**
None afaik, all tests in [crates/swc](https://github.com/swc-project/swc/tree/main/crates/swc) complete successfully and anything that depended on the broken implementation would not have worked with a different transformer.

**Related issue (if exists):**

 - Closes https://github.com/swc-project/swc/issues/5951.

### CODE REVIEW

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thanks!

File: crates/swc_ecma_parser/tests/tsc/awaitCallExpression9_es5.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// @target: ES5
```
Comment (kdy1) [inline crates/swc_ecma_parser/tests/tsc/awaitCallExpression9_es5.ts:1]: Why did you add this file? Please remove

Comment (finnboeger) [reply crates/swc_ecma_parser/tests/tsc/awaitCallExpression9_es5.ts]: To explicitly test the behaviour for arrays, however the current solution doesn't really require it

**Review by kdy1 — APPROVED** [review]
Thank you!

---


swc-bump:
 - swc_ecma_transforms_compat
